lib/mapObjects/MiscObjects.cpp

```diff
@@ -336,7 +336,7 @@ int CGCreature::takenAction(const CGHeroInstance *h, bool allowJoin) const
 	if(count*2 > totalCount)
 		sympathy++; // 2 - hero have similar creatures more that 50%
 
-	int diplomacy = h->valOfBonuses(Bonus::SECONDARY_SKILL_PREMY, SecondarySkill::DIPLOMACY);
+	int diplomacy = h->valOfBonuses(Bonus::CHARISMA_BONUS);
```

Comment:
This code should be rechecked. For example, it does not seems to be matching the wiki:
https://heroes.thelazy.net//index.php/Diplomacy
(at the very least, wiki says that diplomacy has strength threshold, don't see it in code)

*Somebody* is wrong, but not sure whether it is wiki or our code.

Reply:
@IvanSavenko looks like all checks is there. Strength treshold is checked earlier, and other tresholds is checked next. Maybe rewrite it?